src/HEVM.hs

```diff
@@ -200,37 +200,35 @@ makeVmEnv (Behaviour method _ c1 _ _ _ _ _) vm =
 -- | Locate the variables refered to in the act-spec in the vm
 locateStorage :: Ctx -> SolcJson -> Map Id Addr -> Method -> (VM,VM) -> Either StorageLocation StorageUpdate -> (Id, (SMType, SMType))
 locateStorage ctx solcjson contractMap method (pre, post) item =
-  let item' = getLoc item
-      contractId = getContractId item'
-      addr = get contractId contractMap
+  let item' = locFromRewrite item
+      addr = get (contractFromRewrite item) contractMap
 
       Just preContract = view (env . contracts . at addr) pre
       Just postContract = view (env . contracts . at addr) post
 
-
       Just (S _ preValue) = readStorage (view storage preContract) (calculateSlot ctx solcjson item')
       Just (S _ postValue) = readStorage (view storage postContract) (calculateSlot ctx solcjson item')
 
       name :: StorageLocation -> Id
-      name (IntLoc i) = nameFromItem method i
-      name (BoolLoc i) = nameFromItem method i
-      name (BytesLoc i) = nameFromItem method i
+      name (IntLoc   i) = nameFromItem method (i `as` Pre) Pre -- I think these `Pre` are correct..?
```

Comment:
I'm not sure about the `Pre` here? I think the name in this scheme refers to the key in the storage mapping (`Map Id (SMtype, SMType)`), and we then use `timeStore` to turn this into a `Map Id SMType` for the pre / post state as needed, so `nameFromItem` should probably be agnostic to the timing of the `TStorageItem` and just produce the same name regardless?

Reply:
Yes, finally looking at this now, you're definitely right. Fix coming in a minute.